### PR TITLE
Simplenote goes Storyboard

### PR DESCRIPTION
**Pull request by jleandroperez**
## Details:
As per the Advanced Search project, we must drop **NSArrayController** in the Notes List.

Since we'll loose bindings, we **do need** programatical UI initialization. Otherwise, several events might fire before the CoreData Stack has been fully brought up.

In this PR we're moving the main ViewControllers UI to a master Storyboard.

**Apologies about the scale of this PR**. Because of the way the **MainMenu.xib** was built, there is no way to do this in phases. Has to be done in a single shot!

I do promise, though, that the code delta is not massive. Most of the changed LOC(s) belong to the new Storyboard!

/cc'ing a league of extraordinary ladies and gentlemen. Because a risky change calls for awesome friends!

@aerych @emilylaguna @guarani @danielebogo 

Ref. #626

### Scenario: Main UX  ------- cc @aerych **(Thank you!!!)**  👇👇👇👇
- [x] Verify the main UI looks as usual!
- [x] Verify you can Show / Hide the Tags List with **CMD + T**
- [x] Verify that you can Enter / Exit Fullscreen with **Ctrl + CMD + F**
- [x] Verify that clicking over a new Tag yields the associated Notes
- [x] Verify that clicking over any Note causes the Editor to render its contents
- [x] Verify that the Search Field filters the Notes List
- [x] Verify that the New Note button adds a new document to the current selected Tag
- [x] Verify that the New Note button is disabled when you're visualizing the Trash
- [x] Verify that right clicking over a Note (in the Notes List) allows you to Delete a Note
- [x] Verify that right clicking over the Trash folder (in the Tags List) allows you to empty its contents
- [x] Verify that right clicking over a Tag (in the Tags List) allows you to **Rename / Delete** such Tag

### Scenario: Editor  ------- cc @emilylaguna  **(Thank you!!!)**  👇👇👇👇
- [ ] Verify that clicking over the top right `( i )` button displays the Stats for such document
- [ ] Verify that clicking over the top right `( ... )` button displays the Note Actions for such document
- Verify that the Note Actions work as expected (just a quick smoke test!!): 
  - [ ] Pin to Top
  - [ ] Markdown
  - [ ] Share
  - [ ] History
  - [ ] Move to Trash
  - [ ] Publish
  - [ ] Collaborate
- [ ] Verify that when you're visualizing a trashed document, you only get the `Restore Note` action (top right button)
- [ ] Verify that you can Add a new Tag, by typing its name in the Tags Editor
- [ ] Verify that you can remove any Tag (from the bottom Tags Editor)

### Scenario: macOS System Menu (Part 1)  ------- cc @guarani  **(Thank you!!!)**  👇👇👇👇
- [x] Verify that **Simplenote > About Simplenote** opens the About Screen
- [x] Verify that **Simplenote > Privacy Settings** opens the Privacy UI
- [x] Verify that **Simplenote > Empty Trash** empties the Trash Bin
- [x] Verify that **Simplenote > Signout** logs you off!
- [x] Verify that **Simplenote > Quit Simplenote** terminates the app
- [x] Verify that all of the items in the **Note** menu are disabled, whenever you're visualizing a Trashed Item
- [x] Verify that **Note > New Note** adds a new document
- [x] Verify that **Note > Move to Trash** deletes the current document
- [x] Verify that **Note > Print** attempts to print the current document
- [x] Verify that **Edit > Find > Search** highlights the Search Field

### Scenario: macOS System Menu (Part 2)  ------- cc @danielebogo  **(Thank you!!!)**  👇👇👇👇
- [x] Verify that **View > Line Length** lets you modify the Editor's Line Length
- [x] Verify that **View > Font Size** lets you modify the Editor's Font Size (**the editor must be the first responder!**)
- [x] Verify that **View > Note Display** changes the way the Notes List renders
- [x] Verify that **View > Note Sorting** changes the way the Notes List is sorted
- [x] Verify that **View > Tags Sorting** changes the way the Tags List is sorted
- [x] Verify that **View > Theme** lets you change the active Theme
- [x] Verify that **View > Focus Mode** lets you Show / Hide the Notes List and Tags List simultaneously
- [x] Verify that **View > Toggle Sidebar** lets you Show / Hide the Sidebar
- [x] Verify that **View > Enter Fullscreen** lets you enter Fullscreen (obviously!) 😂
- [x] Verify that **Format > Insert Checklist** inserts a new Checklist (**the editor must be the first responder!**)

## Release
These changes do not require release notes.

### CODE REVIEW

**Comment by aerych** [comment]
> Verify that you can Enter / Exit Fullscreen with CMD + F

Should this be `Control + Command + F` ?  Just double checking to be sure.

File: Simplenote/NSStoryboard+Simplenote.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import Foundation
+
+
+// MARK: - NSStoryboard: Helper API(s)
+//
+extension NSStoryboard {
+
+    func instantiateViewController<T: NSViewController>(ofType type: T.Type) -> T {
+        guard let target = instantiateController(withIdentifier: T.sceneIdentifier) as? T else {
```
Comment (aerych) [inline Simplenote/NSStoryboard+Simplenote.swift:9]: Nice :)

File: Simplenote/NoteEditorViewController.m [hunk 2]
```diff
@@ -74,9 +74,9 @@ - (instancetype)initWithCoder:(NSCoder *)coder
     return self;
 }
 
-- (void)awakeFromNib
+- (void)viewDidLoad
 {
-    [super awakeFromNib];
+    [super viewDidLoad];
```
Comment (aerych) [inline Simplenote/NoteEditorViewController.m:79]: Yay for this :)

**Review by danielebogo — APPROVED** [review]
LGTM!
Just one note: When I switch between Theme Dark and Theme System appearance (Dark), the search field background is different. With Dark Theme is actually lighter than the System Dark.

Screenshot 2020-09-01 at 13 28 38

Dark Theme on the left, System Dark on the right

**Comment by jleandroperez** [comment]
@danielebogo I'll follow up on the SearchField in another PR. Thank you for reviewing this one sir!!!

@aerych `delete note` MenuItem's validation fixed (should be disabled in Trash now).

**Thank you both, a loooot, for your time!!**

**Comment by jleandroperez** [comment]
Thanks a million @guarani!!! 😄

**Comment by jleandroperez** [comment]
Merging this PR!! **Thanks everyone!!!**

🔥 🔥 🔥

**Pull request merged**